pkg/apis/configuration/validation/validation.go

```diff
@@ -627,10 +627,18 @@ func validateActionRedirect(redirect *v1.ActionRedirect, fieldPath *field.Path)
 	return allErrs
 }
 
-// Returns a slice of vars enclosed in ${}. For example ${scheme} would return scheme.
-func captureVariables(s string) [][]string {
-	regex := regexp.MustCompile(`\$\{([^}]+)\}`)
-	return regex.FindAllStringSubmatch(s, -1)
+var regexFmtRegexp = regexp.MustCompile(`\$\{([^}]*)\}`)
```

Comment:
`regexFmtRegexp` doesn't describe well the variable. Could you rename it to something like `nginxVariableRegexp`?